src/modules/rollenmapping/api/rollenmapping.controller.ts

```diff
@@ -0,0 +1,173 @@
+import { SchulConnexValidationErrorFilter } from '../../../shared/error/schulconnex-validation-error.filter.js';
+import { RollenmappingCreateBodyParams as RollenmappingCreateBodyParams } from './rollenmapping-create-body.params.js';
+
+@UseFilters(new SchulConnexValidationErrorFilter())
+@ApiTags('rollenmapping')
+@ApiBearerAuth()
+@ApiOAuth2(['openid'])
+@Controller({ path: 'rollenmapping' })
+export class RollenmappingController {
+    public constructor(
+        private readonly rollenMappingRepo: RollenMappingRepo,
+        private readonly rollenMappingFactory: RollenMappingFactory,
+        private readonly logger: ClassLogger,
+    ) {}
+
+    @Get(':id')
+    @ApiOkResponse({ description: 'The rollenmapping was successfully returned', type: RollenMapping })
+    @ApiUnauthorizedResponse({ description: 'Unauthorized to retrieve the rollenmapping' })
+    @ApiNotFoundResponse({ description: 'No rollenmapping found' })
+    @ApiForbiddenResponse({ description: 'Insufficient rights to retrieve the rollenmapping' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while retrieving the rollenmapping' })
+    public async getRollenmappingWithId(
+        @Param('id') id: string,
+        @Permissions() personPermission: PersonPermissions,
+    ): Promise<RollenMapping<true>> {
+        if (!(await personPermission.hasSystemrechteAtRootOrganisation([RollenSystemRecht.ROLLEN_VERWALTEN]))) {
+            throw new ForbiddenException('Insufficient rights to retrieve the rollenmapping objects');
+        }
+
+        const rollenmapping: Option<RollenMapping<true>> = await this.rollenMappingRepo.findById(id);
+
+        if (!rollenmapping) {
+            this.logger.error(`No rollenmapping object found with id ${id}`);
+            throw new NotFoundException(`No rollenmapping object found with id ${id}`);
+        }
+
+        return rollenmapping;
+    }
+
+    @Get()
+    @ApiOkResponse({
+        description: 'The available rollenmapping objects were successfully returned',
+        type: [RollenMapping],
+    })
+    @ApiUnauthorizedResponse({ description: 'Unauthorized to retrieve the rollenmapping' })
+    @ApiNotFoundResponse({ description: 'No rollenmapping objects found' })
+    @ApiForbiddenResponse({ description: 'Insufficient rights to retrieve the rollenmapping objects' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while retrieving the rollenmapping objects' })
+    public async getAllAvailableRollenmapping(
+        @Permissions() personPermission: PersonPermissions,
+    ): Promise<RollenMapping<true>[]> {
+        if (!(await personPermission.hasSystemrechteAtRootOrganisation([RollenSystemRecht.ROLLEN_VERWALTEN]))) {
+            throw new ForbiddenException('Insufficient rights to retrieve the rollenmapping objects');
+        }
+
+        const rollenmappingArray: Option<RollenMapping<true>[]> = await this.rollenMappingRepo.find();
```

Comment:
This needs to be filtered to the Mappings for LMS Systems the User is allowed to edit